legacy/builder/utils/utils.go

```diff
@@ -443,12 +443,21 @@ func CopyDir(src string, dst string, extensions []string) (err error) {
 		return
 	}
 
-	entries, err := ioutil.ReadDir(src)
+	entries, err := os.ReadDir(src)
 	if err != nil {
 		return
 	}
 
+	infos := make([]fs.FileInfo, 0, len(entries))
 	for _, entry := range entries {
+		info, scopeErr := entry.Info()
+		if scopeErr != nil {
+			return
+		}
+		infos = append(infos, info)
+	}
+
+	for _, entry := range infos {
```

Comment:
Looks like we don't need to allocate the array here

```suggestion
	for _, dirEntry := range entries {
		entry, scopeErr := dirEntry.Info()
		if scopeErr != nil {
			return
		}
```

Reply:
I have removed the array allocation block